student.rb

```diff
@@ -1,14 +1,23 @@
-require_relative('./person')
+require_relative './person'
 
 class Student < Person
-  attr_accessor :classroom
+  attr_reader :classroom
 
-  def initialize(age, classroom = 'None', name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
-    @classroom = classroom
+  def initialize(age:, name: 'Unknown', parent_permission: true)
+    super(name: name, age: age, parent_permission: parent_permission)
+    @classroom = []
   end
 
   def play_hooky
     "¯\(ツ)/¯"
   end
+
+  def add_classroom(classroom)
+    if @classroom.include?(classroom)
+      puts 'Student already enrolled in this classroom'
+    else
+      classroom.add_student(self)
+      @classroom.push(classroom)
+    end
+  end
```

Comment:
- [ ] As indicated in the requirements, the relation between student and classroom is a student belongs to... In this case, we want to make sure that the instance class points only to the classroom where the student belongs. So I think we should not use an array here, instead just reference the one classroom.

>Creating the has-many side (a classroom has many students).
Creating the belongs-to side (a student belongs to a classroom).
Making sure that when adding a student to a classroom it also sets the classroom for the student.